### PR TITLE
Add Tests and CI

### PR DESCRIPTION
**Pull request by TheTripleV**

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: Test and Deploy
```
Comment (AustinShalit) [inline .github/workflows/workflow.yml:1]: Just for consistency with the other projects, can you rename the file to `ci.yml` and the workflow name to `ci`?

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+on:
```
Comment (AustinShalit) [inline .github/workflows/workflow.yml:2]: Use the condensed syntax here
```suggestion
on: [pull_request, push]
```

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    branches: 
+      - master
+  push:
+    branches:
+      - master
+  create:
+    tags:
+      - '*'
+jobs:
+  check:
```
Comment (AustinShalit) [inline .github/workflows/workflow.yml:13]: I think `check-format` might be a better job name.

File: .github/workflows/workflow.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Black
+      run: |
+        pip install black
+        black --check --diff .
```
Comment (AustinShalit) [inline .github/workflows/workflow.yml:23]: Separate install from check into 2 different steps please

File: .github/workflows/workflow.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+  test-extension:
```
Comment (AustinShalit) [inline .github/workflows/workflow.yml:25]: Can you add a `name:` option and append the python version to the name? Like this: https://github.com/wpilibsuite/allwpilib/pull/2586/files#diff-e9f950f17198d3d5e3122a44230a09b9R23

File: .github/workflows/workflow.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        run: |
+          python -m pip install -r test-requirements.txt
+      - name: Run Tests for ${{ matrix.python-version }}
+        run: |
+          cd tests
+          python -m pytest -vv
+  pypi-release:
+    needs: test-extension
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'refs/tags/') && github.repository_owner	== 'wpilibsuite'
```
Comment (AustinShalit) [inline .github/workflows/workflow.yml:56]:
```suggestion
    if: startsWith(github.ref, 'refs/tags/v') && github.repository_owner	== 'wpilibsuite'
```

Comment (AustinShalit) [inline .github/workflows/workflow.yml:56]: Should this if statement be only on the publish step - that way we can test building the wheel?

**Comment by TheTripleV** [comment]
I copied this workflow from sphinxext-opengraph. Should the same changes be made there too?

**Comment by AustinShalit** [comment]
Yes, please.

File: .github/workflows/workflow.yml (hunk 6 above)
Comment (TheTripleV) [reply .github/workflows/workflow.yml]: I added building a wheel to the test job so it should be fine behind a guard in pypi-release.

**Comment by TheTripleV** [comment]
I think http -> https is out of the scope of this project as this just extends linkcheck.

Just like linkcheck, linkcheckdiff produces output files in txt and json formats. This could just be parsed in the makefile or in frc-docs's CI.

**Pull request merged**